tests/scripts/thread-cert/Cert_9_2_02_MGMTCommissionerSet.py

```diff
@@ -0,0 +1,183 @@
+        self.nodes[COMMISSIONER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[COMMISSIONER].get_state(), 'router')
+
+        # Skip all other Coaps sent by Leader
+        self.skip_coap_messages(COMMISSIONER, '0.02')
+        self.skip_coap_messages(LEADER, '2.04')
+
+        # Commissioner start
+        self.nodes[COMMISSIONER].commissioner_start()
+        self.simulator.go(3)
+        self.skip_coap_messages(COMMISSIONER, '0.02')    # Skip LEAD_PET.req
+
+        # Get CommissionerSesssionId from LEAD_PET.rsp
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04', assert_enabled=False)
+        commissioner_session_id_tlv = command.get_sub_tlv(msg.coap.payload, mesh_cop.CommissionerSessionId)
+
+        # Step 2 - Harness instructs commissioner to send MGMT_COMMISSIONER_SET.req to Leader
+        steering_data_tlv = mesh_cop.SteeringData(bytes([0xFF]))
+        self.nodes[COMMISSIONER].commissioner_mgmtset_with_tlvs([steering_data_tlv])
+        self.simulator.go(5)
+
+        # Step 3 - Leader responds to MGMT_COMMISSIONER_SET.req with MGMT_COMMISSIONER_SET.rsp
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04')
+        command.check_payload_same(msg.coap.payload, (mesh_cop.State(mesh_cop.MeshCopState.REJECT),)) # (mesh_cop.State(mesh_cop.MeshCopState.REJECT),) <- this a tuple, don't delete the comma
+        self.skip_coap_messages(COMMISSIONER, '0.02')
+
+        # Step 4 - Harness instructs commissioner to send MGMT_COMMISSIONER_SET.req to Leader
+        self.nodes[COMMISSIONER].commissioner_mgmtset_with_tlvs([steering_data_tlv, commissioner_session_id_tlv])
+        self.simulator.go(5)
+        commissioner_messages = self.simulator.get_messages_sent_by(COMMISSIONER)
+        msg = commissioner_messages.next_coap_message('0.02', assert_enabled=False)
+        # Check coap URI and payload
+        assert msg.coap.uri_path == '/c/cs'
+        assert command.contains_tlvs(msg.coap.payload, [mesh_cop.SteeringData, mesh_cop.CommissionerSessionId])
+
+        # Check destination address
+        dest_addr = msg.ipv6_packet.ipv6_header.destination_address
+        rloc = ip_address(self.nodes[LEADER].get_addr_rloc())
+        leader_aloc = ip_address(self.nodes[LEADER].get_addr_leader_aloc())
+        assert dest_addr == rloc or dest_addr == leader_aloc
+
+        # Step 5 - Leader sends MGMT_COMMISSIONER_SET.rsp to commissioner
+        leader_messages = self.simulator.get_messages_sent_by(LEADER)
+        msg = leader_messages.next_coap_message('2.04')
+        command.check_payload_same(msg.coap.payload, (mesh_cop.State(mesh_cop.MeshCopState.ACCEPT),))
+
+        # Step 6 - Leader sends a multicast MLE Data Response
+        msg = leader_messages.next_mle_message(mle.CommandType.DATA_RESPONSE)
+        command.check_data_response(msg, command.NetworkDataCheck(
```

Comment:
I would prefer previous `check_data_response()` style by simply extending the `check_network_data()`.
e.g. add check_type (NetworkDataCheckType.COMM_DATASET), and the `check_datail[1] `would be the sub tlv type list.